File: benchmarks/sieve/bench.py
import time
def sieve(n):
    is_prime = bytearray([1]) * (n + 1)
    is_prime[0] = 0; is_prime[1] = 0
    i = 2
    while i * i <= n:
        if is_prime[i]:
            j = i * i
            while j <= n: is_prime[j] = 0; j += i
        i += 1
    return sum(is_prime)
N = 10_000_000
t = time.perf_counter()
c = sieve(N)
print(f"Primes up to {N}: {c} ({(time.perf_counter()-t)*1000:.1f}ms)")

File: benchmarks/sieve/bench.rs
use std::time::Instant;
fn sieve(n: usize) -> usize {
    let mut is_prime = vec![true; n + 1];
    is_prime[0] = false; is_prime[1] = false;
    let mut i = 2;
    while i * i <= n { if is_prime[i] { let mut j = i*i; while j <= n { is_prime[j] = false; j += i; } } i += 1; }
    is_prime.iter().filter(|&&x| x).count()
}
fn main() {
    let n = 10_000_000usize;
    let t = Instant::now();
    let c = sieve(n);
    println!("Primes up to {}: {} ({}ms)", n, c, t.elapsed().as_millis());
}
